### DISCUSSION TITLE
Specify regex when parsing folder names for metadata

### DISCUSSION THREAD

**Discussion by zalavariandris** [jellyfin/jellyfin#10343] [Q&A]
Hi!

I would like to use Jellyfin with my current folder structure.
I store my movie files in folders which include the director. It looks like this: {Director} - {Title} ({Year}).
I like having the directors in there. It helps me remember who did what.

My Problem is that Jellyfin does not recognize movies named this way, and I don't want to fix the metadata for all the movies, nor renaming my folders.
Is there a way to provide jellyfin a regex for example to identify the movie when the director is present? (I have seen something similar in the code around here: https://github.com/jellyfin/jellyfin/blob/40f7eb4e8cd9e250fb3870a4799a5a8d949e2068/Emby.Naming/Video/CleanDateTimeParser.cs) I am not sure how it's used elswhere, nor if its currently accessable from the UI)

**Comment by thevortexcloud** [comment]
Jellyfin does have a folder view which may partially do what you want.

**Comment by zalavariandris** [comment]
Not really. I want jellyfin to parse my directory structure and extract metadata from folder names. It is already able to recognize the movie if I name my folder in a specific way. e.g.: _An awesome movie title (2023)_. But if I have the director in the folder name it wont recognize the movie.
By looking at the code it does seem to have a function to parse folder names with a regex. I would like to specify a regex that is able to recognize movies with another pattern.
